index.js

```diff
@@ -50,17 +47,49 @@ exports.get = function(req, res) {
             } else {
                 filestr = file;
             }
+            let rstr = '';
+            if (currentDir !== dir) {
+                rstr = currentDir;
+            }
             data.push({
                 Name : filestr,
                 Ext : ext,
                 IsDirectory: false,
-                Path : path.join(query, file)
+                Path : path.join(query, file),
+                Root : rstr
             });
         }
-      });
-      data = _.sortBy(data, function(f) { return f.Name });
-      res.json(data);
-  });
+    });
+    return data;
+}
+
+function readRoots(roots, res, query, fullList) {
+    let currentDir = roots.shift();
+
+    fs.readdir(currentDir, function (err, files) {
+        let data = fullList.concat(displayFiles(err, files, currentDir, query));
+
+        if (roots.length > 0) {
+            readRoots(roots, res, query, data);
```

Comment:
Brenden 'Friar' Tuck [6:22 PM]
nothing pops out at me.
I would mark readRoots as recursive, though, just to make it obvious.

davek [8:04 PM]
you mean in a comment?  Or is there a keyword I don't know about?

(I was trying to avoid the pyramid of doom)

Brenden 'Friar' Tuck [8:11 PM]
just a comment was all I was thinking.
recursive async code can get gnarly.  reminders help.

Reply:
fixed

Reply:
fixed